woodwork/tests/type_system/test_ltype_inference.py

```diff
@@ -125,6 +133,32 @@ def test_natural_language_inference(natural_language):
             assert isinstance(inferred_type, NaturalLanguage)
 
 
+@patch("woodwork.type_sys.inference_functions.natural_language_func")
+def test_nl_inference_called_on_no_other_matches(nl_mock, pandas_natural_language):
+    new_type_sys = TypeSystem(
+        inference_functions=DEFAULT_INFERENCE_FUNCTIONS,
+        relationships=DEFAULT_RELATIONSHIPS,
+        default_type=DEFAULT_TYPE,
+    )
+    new_type_sys.inference_functions[NaturalLanguage] = nl_mock
+    inferred_type = new_type_sys.infer_logical_type(pandas_natural_language[0])
+    assert isinstance(inferred_type, NaturalLanguage)
+    assert nl_mock.called
+
+
+@patch("woodwork.type_sys.inference_functions.natural_language_func")
+def test_nl_inference_not_called_with_other_matches(nl_mock, pandas_integers):
```

Comment:
a suggestion for a third version of this test would be to confirm that nl inference function had been called on a series whose logical type is inferred as Unknown

Reply:
Added this test. Also had to rework the test a little bit to remove some assertions since the mocked function doesn't return the right value. Just was a happy accident that things worked with the tests as they were before.